src/StringTools/FowlerNollVo1aHash.cs

```diff
@@ -0,0 +1,135 @@
+
+        /// <summary>
+        /// Computes 32 bit Fowler/Noll/Vo-1a hash of a string (regardless of encoding).
+        /// </summary>
+        /// <param name="text">String to be hashed.</param>
+        /// <returns>32 bit signed hash</returns>
+        public static int ComputeHash32(string text)
+        {
+            uint hash = fnvOffsetBasisA32Bit;
+
+#if NET35
+            unchecked
+            {
+                for (int i = 0; i < text.Length; i++)
+                {
+                    char ch = text[i];
+                    byte b = (byte)ch;
+                    hash ^= b;
+                    hash *= fnvPrimeA32Bit;
+
+                    b = (byte)(ch >> 8);
+                    hash ^= b;
+                    hash *= fnvPrimeA32Bit;
+                }
+            }
+#else
+            ReadOnlySpan<byte> span = MemoryMarshal.Cast<char, byte>(text.AsSpan());
+            foreach (byte b in span)
+            {
+                hash = unchecked((hash ^ b) * fnvPrimeA32Bit);
+            }
+#endif
+
+            return unchecked((int)hash);
+        }
+
+        /// <summary>
+        /// Computes 64 bit Fowler/Noll/Vo-1a hash optimized for ASCII strings.
+        /// The hashing algorithm considers only the first 8 bits of each character.
+        /// Analysis: https://github.com/KirillOsenkov/MSBuildStructuredLog/wiki/String-Hashing#faster-fnv-1a
+        /// </summary>
+        /// <param name="text">String to be hashed.</param>
+        /// <returns>64 bit unsigned hash</returns>
+        public static long ComputeHash64Fast(string text)
+        {
+            long hash = fnvOffsetBasisA64Bit;
+
+            unchecked
+            {
+                for (int i = 0; i < text.Length; i++)
+                {
+                    char ch = text[i];
+
+                    hash = (hash ^ ch) * fnvPrimeA64Bit;
+                }
+            }
+
+            return hash;
+        }
+
+        /// <summary>
+        /// Computes 64 bit Fowler/Noll/Vo-1a hash of a string (regardless of encoding).
+        /// </summary>
+        /// <param name="text">String to be hashed.</param>
+        /// <returns>64 bit unsigned hash</returns>
+        public static long ComputeHash64(string text)
+        {
+            long hash = fnvOffsetBasisA64Bit;
+
+#if NET35
+            unchecked
+            {
+                for (int i = 0; i < text.Length; i++)
+                {
+                    char ch = text[i];
+                    byte b = (byte)ch;
+                    hash ^= b;
+                    hash *= fnvPrimeA64Bit;
+
+                    b = (byte)(ch >> 8);
+                    hash ^= b;
+                    hash *= fnvPrimeA64Bit;
+                }
+            }
+#else
+            ReadOnlySpan<byte> span = MemoryMarshal.Cast<char, byte>(text.AsSpan());
+            foreach (byte b in span)
+            {
+                hash = unchecked((hash ^ b) * fnvPrimeA64Bit);
+            }
```

Comment:
@KalleOlaviNiemitalo - Thank you for bringing this up! 
We're not declaring what can be the expectations of stableness of the hashing between different systems (we actually do not declare anything ... yet :-)) - but it'd certainly be nice to have it more universal.

@uweigand - would you be willing to chime in and share your thoughts on the following?

Basically - since the `MemoryMarshal.Cast<char, byte>` changes how we fetch data from memory to processor - it'll behave differently between Endinan-ness representations. But how anout the version which manually splits each char to bytes:
```csharp
for (int i = 0; i < text.Length; i++)
{
    char ch = text[i];
    byte b = (byte)ch;
    hash ^= b;
    hash *= fnvPrimeA64Bit;

    b = (byte)(ch >> 8);
    hash ^= b;
    hash *= fnvPrimeA64Bit;
}
```
since all this happens in processor after the data are fetched from memory - it should fetch same results regardless of endinan-ness - is that right?

Reply:
Thank you for all the detailed input! I'll remove the Spans version.

The Spans version is indeed not superior (it has slight to no advantage for large strings, but performs worse for a very short strings). It's nice that it doesn't need unchecked code and is more brief - but given the fact that it can perform worse and that the other version anyways need to be maintained - it's an easy choice.

I'll as well fix the description for the `ComputeHash64Fast`